sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -239,5 +247,777 @@ public virtual Response<TimeSeriesModelSettings> UpdateModelSettingsDefaultTypeI
+        /// </summary>
+        /// <param name="timeSeriesNames">List of names of the Time Series instance to return.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// List of instance or error objects corresponding by position to the array in the request. Instance object is set when operation is successful
+        /// and error object is set when operation is unsuccessful.
+        /// </returns>
+        /// <seealso cref="GetInstancesAsync(IEnumerable{string}, CancellationToken)">
+        /// See the asynchronous version of this method for examples.
+        /// </seealso>
+        public virtual Response<InstancesOperationResult[]> GetInstances(
+            IEnumerable<string> timeSeriesNames,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNullOrEmpty(timeSeriesNames, nameof(timeSeriesNames));
+
+                InstancesBatchRequest batchRequest = new InstancesBatchRequest()
+                {
+                    Get = new InstancesRequestBatchGetOrDelete()
+                };
+
+                foreach (string timeSeriesName in timeSeriesNames)
+                {
+                    batchRequest.Get.Names.Add(timeSeriesName);
+                }
+
+                Response<InstancesBatchResponse> executeBatchResponse = _timeSeriesInstancesRestClient
+                    .ExecuteBatch(batchRequest, _clientSessionId, cancellationToken);
+
+                return Response.FromValue(executeBatchResponse.Value.Get.ToArray(), executeBatchResponse.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets Time Series instances from the environment by Time Series Id asynchronously.
+        /// </summary>
+        /// <param name="timeSeriesIds">List of Ids of the Time Series instance to return.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// List of instance or error objects corresponding by position to the array in the request. Instance object is set when operation is successful
+        /// and error object is set when operation is unsuccessful.
+        /// </returns>
+        /// <remarks>
+        /// For more samples, see <see href="https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/samples">our repo samples</see>.
+        /// </remarks>
+        /// <example>
+        /// <code snippet="Snippet:TimeSeriesInsightsSampleGetInstancesByIds">
+        /// </code>
```

Comment:
snippet

Reply:
will eventually be populated when the sample is written.